addons/account_edi/models/account_edi_document.py

```diff
@@ -52,10 +52,10 @@ def _compute_edi_content(self):
                 config_errors = doc.edi_format_id._check_move_configuration(move)
                 if config_errors:
                     res = base64.b64encode('\n'.join(config_errors).encode('UTF-8'))
-                elif move.is_invoice(include_receipts=True) and doc.edi_format_id._is_required_for_invoice(move):
-                    res = base64.b64encode(doc.edi_format_id._get_invoice_edi_content(doc.move_id))
-                elif move.payment_id and doc.edi_format_id._is_required_for_payment(move):
-                    res = base64.b64encode(doc.edi_format_id._get_payment_edi_content(doc.move_id))
+                else:
+                    move_applicability = doc.edi_format_id._get_move_applicability(move)
+                    if move_applicability and move_applicability.get('edi_content'):
+                        res = base64.b64encode(getattr(doc.edi_format_id, move_applicability['edi_content'])(move))
```

Comment:
Nope, that is completely unnecessary, have `_get_move_applicability` return callbacks if it wants to make shit callable. It literally already has all the information it needs already so it can even return parameter-less callbacks created with `functools.partial`.

Reply:
Indeed... We got rid of all the new getattr in this PR, but unfortunately we still need your review on the enterprise part because of existing gettatr which were indented. Could you have a look please ?